### ISSUE TITLE
LSP: The callback for workspace/applyEdit does not send a response

### ISSUE THREAD

**Issue by Jesse-Bakker** [neovim/neovim#11608]
- `nvim --version`:
```
NVIM v0.5.0-261-g2ef72437f
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/gcc-5 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -g -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/home/travis/build/neovim/bot-ci/build/neovim/build/config -I/home/travis/build/neovim/bot-ci/build/neovim/src -I/home/travis/build/neovim/bot-ci/build/neovim/.deps/usr/include -I/usr/include -I/home/travis/build/neovim/bot-ci/build/neovim/build/src/nvim/auto -I/home/travis/build/neovim/bot-ci/build/neovim/build/include
Compiled by travis@travis-job-15cad2f4-d0e5-45e9-9231-17653111c5a0

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/share/nvim"

Run :checkhealth for more info
```
- `vim -u DEFAULTS` (version: ) behaves differently?
It does not have the lsp api
- Operating system/version: 
Linux 5.4.6-arch1-1 x86_64 GNU/Linux (Arch Linux)
- Terminal name/version: 
alacritty 0.4.0
- `$TERM`:
`xterm-256color`

### Steps to reproduce using `nvim -u NORC`
Using my PR #11607 applying any code action that returns workspace/applyEdit,
```
nvim -u NORC
:lua vim.lsp.buf.code_action()
```

### Actual behaviour
Error log with message: `Error executing vim.schedule lua callback: /usr/share/nvim/runtime/lua/vim/lsp/rpc.lua:360: method "workspace/applyEdit": either a result
or an error must be sent to the server in response`
The edit however is applied
### Expected behaviour
No error

**Comment by norcalli** [comment]
Ah yes apparently that does require returning something. When I modified that from the original PR, I guess I never bothered to check if it was correct. I read the spec and I think the following change might work. Try applying it and see how it goes.

```diff
diff --git a/runtime/lua/vim/lsp/callbacks.lua b/runtime/lua/vim/lsp/callbacks.lua
index b69fb4464..e409c985e 100644
--- a/runtime/lua/vim/lsp/callbacks.lua
+++ b/runtime/lua/vim/lsp/callbacks.lua
@@ -17,7 +17,11 @@ M['workspace/applyEdit'] = function(_, _, workspace_edit)
   if workspace_edit.label then
     print("Workspace edit", workspace_edit.label)
   end
-  util.apply_workspace_edit(workspace_edit.edit)
+  local status, err = pcall(util.apply_workspace_edit, workspace_edit.edit)
+  return {
+    applied = status;
+    failureReason = err;
+  }
 end
 
 M['textDocument/publishDiagnostics'] = function(_, _, result)
```

If it works, you can add it to that PR.

**Comment by Jesse-Bakker** [comment]
That works. I have added it to the PR

**Comment by Jesse-Bakker** [comment]
Fixed by #12270
